Changes for Mirth 3.12 connectivity

rogin opened the pull request:
Changes were broken into multiple areas. Feel free to cherry-pick.

rogin committed:
Bug fixes and refactor for PS 6+

-bump version
-require PS v6+
-use proper psd1 filename
-enable property to skip cert check, remove PS v5 cert solution
-refactor Mirth login to omit creds in URL
-add required header to API calls
-fix XML node selection bugs in Get-MirthChannels
-fix buggy param usage in Import-MirthChannel

rogin committed:
cleanup

-remove unused $VERSION
-resolve plaintext password warnings
-create Save-Content to simplify saving responses

rogin committed:
update test to reflect module version info

rogin committed:
use securestring for mirth logins

rogin committed:
Refactor interaction with config options

-revert removal of Add-Type to fix module import error
-refactor reading and setting config options
-remove "global" from functions
-replace usage of "-ContentType" with headers

rogin committed:
Fix API output to file

-omit ending slash for output folder
-Cleanup switch declarations
-Fix API output to file

rogin committed:
expand options of module import during testing

rogin committed:
Move required assemblies, alter aliases export

rogin committed:
create alias for Connect-Mirth

rogin committed:
return cleaner hashtable for Server About

rogin committed:
fix error in output saving

naql reviewed (commented):
Going to take a while to get through all of that, I"m about a third of the way thru the psm1 changes.  Looks mostly like cleanup. 

I"ll chat with you in DataSprite about it as I circle back.

Thanks for the commits!

rogin committed:
enable formatter

rogin committed:
have channel statuses return a clean array

rogin committed:
add mirth logout

rogin committed:
enable autocompletion of channel info

-adds Get-MirthChannelIdsAndNames

rogin committed:
Refactor to multiple files

-add more functions to return hashtable results

rogin committed:
add pester tests

-fix function export in PSD1
-force module reload when running tests
-refactor to use new ConvertFrom-Xml

rogin committed:
correct "accept" header

rogin committed:
Always return an array for a 1-item result

rogin committed:
Simplify XML conversions for lists and maps

rogin committed:
changes to pester tests

rogin committed:
Fix saving of XMl elements

rogin committed:
Avoid output of AppendChild

rogin committed:
Refactor credential usage

-add param set for host and port from config files

rogin committed:
Refactor XML to hashtable conversions

-add more tests

rogin committed:
add default credentials

rogin committed:
Implement message searching

rogin committed:
Expand examples

rogin committed:
Allow channel name to ID lookups

rogin committed:
Make Save-Content public

rogin committed:
Add calls for licensemanager extension

rogin committed:
Fix param name and debug statements

rogin committed:
Add call for offline license validation

rogin committed:
Add database endpoints

rogin committed:
add events endpoints

rogin committed:
add extensions endpoints

rogin committed:
move to more accurate folder

rogin committed:
Fix quirk when finding usable child nodes

rogin committed:
fix XML output when given an XmlDocument

rogin committed:
fix login test

rogin committed:
reorg folders, add a dozen endpoints

rogin committed:
add more endpoints

rogin committed:
fix the XML response conversions

tonygermano commented:
@rogin I'd recommend closing this PR and breaking it up into many smaller ones that can be merged separately. I'd only have multiple commits in the same PR if they are related and dependent on one another. This is overwhelming to look at.

naql commented:
> @rogin I'd recommend closing this PR and breaking it up into many smaller ones that can be merged separately. I'd only have multiple commits in the same PR if they are related and dependent on one another. This is overwhelming to look at.

Since his supports a newer version of Mirth than mine does, I'm going to end up just merging it in entirely, I think, and then just go about tweaking what I need to, to work with my deploy script.

rogin committed:
allow registration of channelId autocompletion

rogin committed:
improve conversion debug

rogin committed:
access login creds in standard way

rogin committed:
new command for enhanced data query
